pandas/tests/resample/test_datetime_index.py

```diff
@@ -1135,6 +1135,15 @@ def test_resample_nunique():
     assert_series_equal(result, expected)
 
 
+def test_resample_nunique_preserves_column_level_names():
+    # see gh-23222
+    df = tm.makeTimeDataFrame(freq="1D").abs()
+    df.columns = pd.MultiIndex.from_arrays([df.columns.tolist()] * 2,
+                                           names=["lev0", "lev1"])
+    result = df.resample("1h").nunique()
+    tm.assert_index_equal(df.columns, result.columns)
```

Comment:
This test might as well compare a `result` and `expected` dataframe.

Reply:
The expected `DataFrame` has 697 rows.